core/src/main/scala/org/bitcoins/core/protocol/script/ScriptPubKey.scala

```diff
@@ -866,6 +866,70 @@ object NonStandardNotIfConditionalScriptPubKey
   }
 }
 
+sealed trait P2PKWithTimeoutScriptPubKey extends RawScriptPubKey {
+
+  val pubKey: ECPublicKey =
+    ECPublicKey.fromBytes(asm(2).bytes)
+
+  val lockTime: ScriptNumber = ScriptNumber.fromBytes(asm(5).bytes)
+
+  val timeoutPubKey: ECPublicKey =
+    ECPublicKey.fromBytes(asm(9).bytes)
+}
+
+object P2PKWithTimeoutScriptPubKey
+    extends ScriptFactory[P2PKWithTimeoutScriptPubKey] {
+  private case class P2PKWithTimeoutScriptPubKeyImpl(asm: Vector[ScriptToken])
+      extends P2PKWithTimeoutScriptPubKey
+
+  override def fromAsm(asm: Seq[ScriptToken]): P2PKWithTimeoutScriptPubKey = {
+    buildScript(
+      asm = asm.toVector,
+      constructor = P2PKWithTimeoutScriptPubKeyImpl.apply,
+      invariant = isP2PKWithTimeoutScriptPubKey,
+      errorMsg = s"Given asm was not a P2PKWithTimeoutScriptPubKey, got $asm"
+    )
+  }
+
+  def apply(
+      pubKey: ECPublicKey,
+      lockTime: ScriptNumber,
+      timeoutPubKey: ECPublicKey): P2PKWithTimeoutScriptPubKey = {
+    val timeoutAsm = CLTVScriptPubKey(lockTime, EmptyScriptPubKey).asm.toVector
+    val pubKeyAsm = BitcoinScriptUtil
+      .calculatePushOp(pubKey.bytes)
+      .toVector ++ Vector(ScriptConstant(pubKey.bytes))
+    val timeoutPubKeyAsm = BitcoinScriptUtil
+      .calculatePushOp(timeoutPubKey.bytes)
+      .toVector ++ Vector(ScriptConstant(timeoutPubKey.bytes))
+
+    P2PKWithTimeoutScriptPubKeyImpl {
```

Comment:
This is different, `{}` rather than `()`? I didn't know this is valid scala syntax for constructors.

Reply:
Blocks `{}` are allowed anywhere, I don't believe there is any real difference between `{}` and `()` other than formatting

Reply:
I would prefer to just stick the normal syntax unless there is good reason to chagne